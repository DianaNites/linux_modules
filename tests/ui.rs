#![allow(unused_imports, unused_mut, unreachable_code)]
use std::str::from_utf8;

use anyhow::Result;
use assert_cmd::{prelude::*, Command};
use insta::assert_snapshot;

/// amdgpu is a complicated module that makes use of many features,
/// all of which we want to display correctly.
///
/// This test will inherently have unstable output, the goal is to make sure
/// the formatting is correct, not the contents.
// TODO: Figure out `insta` filters to ignore changing contents
#[test]
#[ignore = "Flaky"]
fn amdgpu_info() -> Result<()> {
    let mut cmd = Command::cargo_bin("nms")?;
    cmd //.
        .env_clear()
        .env("COLUMNS", "120")
        .args(["info", "amdgpu"]);
    let out = cmd.output()?;
    let out = from_utf8(&out.stdout)?;

    assert_snapshot!(out);

    Ok(())
}

#[test]
fn help() -> Result<()> {
    for arg in [
        &["--help"][..],
        &["list", "--help"][..],
        &["insert", "--help"][..],
        &["remove", "--help"][..],
        &["info", "--help"][..],
    ] {
        let mut cmd = Command::cargo_bin("nms")?;
        cmd.env_clear().args(arg);
        let out = cmd.output()?;
        let out = from_utf8(&out.stdout)?;

        assert_snapshot!(out);
    }

    Ok(())
}

#[test]
#[ignore = "Flaky"]
fn list() -> Result<()> {
    let mut cmd = Command::cargo_bin("nms")?;
    cmd.env_clear().env("COLUMNS", "120").args(["list"]);
    let out = cmd.output()?;
    let out = from_utf8(&out.stdout)?;

    assert_snapshot!(out);

    Ok(())
}
