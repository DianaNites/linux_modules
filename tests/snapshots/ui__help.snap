---
source: tests/ui.rs
expression: out
---
linux_modules 0.3.6
Manage Linux Kernel Modules

USAGE:
    nms -q
    nms <SUBCOMMAND>

OPTIONS:
    -h, --help       Print help information
    -V, --version    Print version information

SUBCOMMANDS:
    completions    Generate shell completions, outputting to stdout
    info           Get information on a kernel module
    insert         Load a kernel module
    list           List loaded kernel modules
    remove         Remove/unload a kernel module

